apps/hash-ai-worker-ts/src/activities/shared/get-flow-context.ts

```diff
@@ -0,0 +1,108 @@
+import { createGraphClient } from "@local/hash-backend-utils/create-graph-client";
+import { createTemporalClient } from "@local/hash-backend-utils/temporal";
+import { parseHistoryItemPayload } from "@local/hash-backend-utils/temporal/parse-history-item-payload";
+import type { GraphApi } from "@local/hash-graph-client";
+import { getRequiredEnv } from "@local/hash-isomorphic-utils/environment";
+import type { RunFlowWorkflowParams } from "@local/hash-isomorphic-utils/flows/temporal-types";
+import type {
+  AccountId,
+  EntityId,
+  EntityUuid,
+  OwnedById,
+} from "@local/hash-subgraph";
+import { entityIdFromComponents } from "@local/hash-subgraph";
+import { Context } from "@temporalio/activity";
+
+import { logToConsole } from "../../shared/logger";
+
+const temporalClient = await createTemporalClient();
+
+const getWorkflowExecutionStartedEventAttributesOfWorkflow = async (params: {
+  workflowId: string;
+}) => {
+  const { workflowId } = params;
+
+  const handle = temporalClient.workflow.getHandle(workflowId);
+
+  const { events } = await handle.fetchHistory();
+
+  if (!events) {
+    throw new Error(`No events found for workflowId ${workflowId}`);
+  }
+
+  const workflowExecutionStartedEventAttributes =
+    events.find((event) => event.workflowExecutionStartedEventAttributes)
+      ?.workflowExecutionStartedEventAttributes ?? undefined;
+
+  return { workflowExecutionStartedEventAttributes };
+};
+
+type FlowContext = {
+  flowEntityId: EntityId;
+  graphApiClient: GraphApi;
+  userAuthentication: { actorId: AccountId };
+  webId: OwnedById;
+};
+
+const graphApiClient = createGraphClient(logToConsole, {
+  host: getRequiredEnv("HASH_GRAPH_API_HOST"),
+  port: parseInt(getRequiredEnv("HASH_GRAPH_API_PORT"), 10),
+});
```

Comment:
Couldn't this just be exported from this file? It's not specific to the Flow. I guess it only makes a difference if there are callers which _only_ need this client (in which case there's no need to wait for the event history to be searched to give it to them). If we add a cache there's no problem with keeping it in the same context function

Reply:
I've moved it into a standalone file where it can be imported from in https://github.com/hashintel/hash/pull/4379/commits/e8ae052ca1e7da509f204959d88dfb152b137b3f